mvlearn/decomposition/base.py

```diff
@@ -77,3 +84,202 @@ def fit_transform(self, Xs, y=None):
+        y : ignored
+
+        Returns
+        -------
+        self: returns an instance of self.
+        """
+        Xs, n_views, n_samples, n_features = check_Xs(
+            Xs, copy=False, return_dimensions=True
+        )
+        self.n_views_ = n_views
+        self.n_features_ = n_features
+        self.n_samples_ = n_samples
+        if self.n_components is not None:
+            self.pcas_ = ViewTransformer(PCA(n_components=self.n_components))
+
+        if self.n_components is not None:
+            reduced_X = self.pcas_.fit_transform(Xs)
+        else:
+            reduced_X = Xs.copy()
+        reduced_X = np.array(reduced_X)
+        unmixings_, S = self.fit_reduced(reduced_X)
+        mixing_ = np.array([np.linalg.pinv(W) for W in unmixings_])
+        self.components_ = unmixings_
+        self.mixing_ = mixing_
+        if self.n_components is not None:
+            pca_components = []
+            for i, estimator in enumerate(self.pcas_.estimators_):
+                K = estimator.components_
+                pca_components.append(K)
+            self.pca_components_ = np.array(pca_components)
+
+        if self.n_components is None:
+            self.individual_components_ = unmixings_
+            self.individual_mixing_ = mixing_
+        else:
+            self.individual_mixing_ = []
+            self.individual_components_ = []
+            sources_pinv = linalg.pinv(S)
+            for x in Xs:
+                lstq_solution = np.dot(sources_pinv, x)
+                self.individual_components_.append(
+                    linalg.pinv(lstq_solution).T
+                )
+                self.individual_mixing_.append(lstq_solution.T)
+        return self
+
+    def transform(self, Xs, index=None):
+        r"""
+        Recover the sources from each view (apply unmixing matrix).
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+            Training data to recover a source and unmixing matrices from.
+
+        index: int or array-like, default=None
+            The index or list of indices of the fitted views to which the
+            inputted views correspond. If None, there should be as many
+            inputted views as the fitted views and in the same order.
+            Note that the index parameter is not available in all methods of
+            mvlearn yet.
+
+        Returns
+        -------
+        Xs_new : numpy.ndarray, shape (n_views, n_samples, n_components)
+            The mixed sources from the single source and per-view unmixings.
+        """
+
+        if not hasattr(self, "components_"):
+            raise ValueError("The model has not yet been fitted.")
+
+        if index is None:
+            index_ = np.arange(self.n_views_)
+        else:
+            index_ = np.copy(index)
+            index_ = np.atleast_1d(index_)
```

Comment:
This should probably be the case everywhere? asserting at least 1D. Checking the index seems to be a highly repeated set of calls but not sure if it's reasonable to make a related utility function.

Reply:
Yes I believe I also do it in GroupICA, GroupPCA and BaseTransformer. 
I can make it a utility function if you think it is a good idea.

Reply:
I can do it later when I add it to other methods. But i think `np.atleast_1d(index_)` might be useful elsewhere here?

Reply:
I am not sure I understand I think it is an English issue. Do you mean that I forgot to put np.atleast_1d(index) at some other place ?

Reply:
You are correct. Thanks!